IDL/annotations.idl

```diff
@@ -0,0 +1,56 @@
+#include "helpers/basic_inner_types.idl"
+
+@annotation AnnotationTest
+{
+    short var_short;
+    unsigned short var_ushort;
+    long var_long;
+    unsigned long var_ulong;
+    long long var_longlong;
+    unsigned long long var_ulonglong;
+    float var_float;
+    double var_double;
+    long double var_longdouble;
+    boolean var_boolean;
+    octet var_octet;
+    char var_char8;
+    wchar var_char16;
+    string var_string;
+    wstring var_wstring;

```

Comment:
Maybe we should consider declaring within an enumeration and using it as it is done in some of the builtin annotations. In fact, the same name as the enumeration defined in the `basic_inner_types.idl` can be used in order to check the following condition from the specification:

> Enumerations, constants, and typedefs declared within the annotation body may be used unscoped subsequently in the annotation body or further when applying the annotation. They are not significant anywhere else.

As an example of an enumeration declaration within an annotation declaration, the following builtin annotation:

```
@annotation autoid {
    enum AutoidKind {
        SEQUENTIAL,
        HASH
        };
    AutoidKind value default HASH;
    };
```

Reply:
added in https://github.com/eProsima/dds-types-test/pull/6/commits/57ec2286866c7628f3819fc1e9a58f970feabce6 and https://github.com/eProsima/dds-types-test/pull/6/commits/eed0a57daa88d8185d49827aed5001c532a4898c